tests/console/generate_dud.pm

```diff
@@ -0,0 +1,39 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Generate DUD dynamically using mkdud and xml file where variables
+# are expanded for the corresponding product/build.
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use utils qw(zypper_call);
+use autoyast qw(expand_variables);
+use registration qw(add_suseconnect_product get_addon_fullname);
+
+sub run {
+    select_console 'root-console';
+
+    my $xml = 'add_on_products.xml';
+    my $dud = get_required_var('DUD');
+
+    add_suseconnect_product(get_addon_fullname('phub'));
+    zypper_call('in mkdud');
+    my $content = expand_variables(get_test_data($xml));
+    save_tmp_file($xml, $content);
+    assert_script_run('wget -P inst-sys ' . data_url($xml));
+    assert_script_run("mkdud --create $dud --dist sle15 " .
+          "--install instsys,repo --obs-keys --name 'Update' inst-sys");
+    upload_asset($dud);
+}
+
```

Comment:
good to have a post_fail here. wdyt?

Reply:
Could you please be more specific what do you want to add in a post_fail which is not in opensusebasetest?

Reply:
Not sure. i was thinking about the upload the `$xml` and `$dud`. i saw  `add_suseconnect_product` and `mkdud` command and i started thinking that this might need some extra consideration. `$dud` can be found in the assets but if it failed before the upload we miss it. However i am not sure if what exactly the post_fail will provide in this case. Checking some relevant other module i can see that are missing from the default post_fail is the `df` output and the zypp logs. But at the end you are right. the default post_fail might be enough.